lib/smb2/file.rb

```diff
@@ -119,4 +165,51 @@ def size
     info.end_of_file
   end
 
+  # Write the entire contents of `data`, starting at `offset` bytes from the
+  # beginning of the file.
+  #
+  # @param data [String] what to write
+  # @param offset [Fixnum] where in the file to start writing
+  # @return [Fixnum] number of bytes written. This may be less than the length
+  #   of `data` if there was an error.
+  def write(data, offset: self.pos)
+    max = tree.client.max_write_size
+    bytes_written = 0
+    seek(offset)
+
+    while data.length > bytes_written
+      data_chunk = data.slice(bytes_written, max)
```

Comment:
`#each_slice` is no longer a method on Strings since they are not Enumerable after 1.9